error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/MockitoMockClassReference.java

```diff
@@ -60,18 +68,28 @@ public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState
       return Description.NO_MATCH;
     }
 
-    // XXX: Add similar matchers for usage in a ReturnTree and the method's return type.
     Tree parent = state.getPath().getParentPath().getLeaf();
     if (parent.getKind() == VARIABLE
         && INCOMPATIBLE_VARIABLE.matches((VariableTree) parent, state)) {
       return Description.NO_MATCH;
+    } else if (parent.getKind() == RETURN
+        && INCOMPATIBLE_RETURN.matches((ReturnTree) parent, state)) {
+      return Description.NO_MATCH;
     }
 
     return describeMatch(tree, SuggestedFix.delete(Iterables.getOnlyElement(tree.getArguments())));
   }
 
   private static boolean hasTypeDifference(VariableTree tree, VisitorState state) {
-    return !ASTHelpers.isSameType(
-        ASTHelpers.getType(tree), ASTHelpers.getType(tree.getInitializer()), state);
+    return hasTypeDifference(tree, tree.getInitializer(), state);
+  }
+
+  private static boolean hasTypeDifference(ReturnTree tree, VisitorState state) {
+    Tree returnTypeTree = requireNonNull(state.findEnclosing(MethodTree.class)).getReturnType();
```

Comment:
This `requireNonNull` is safe because passed is a `ReturnTree`. Alternatively we could have passed the matching `MethodInvocationTree` (to prevent having to do `tree.getExpression` again), but then finding the enclosing `MethodTree` is less 'safe' in case this method gets reused. I found this to be cleaner and clearer given the similar method signatures.

Furthermore, inlining as a lambda expression instead was [flagged by ErrorProne](https://errorprone.info/bugpattern/UnnecessaryLambda), and indeed I do find this to read better.